bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/scanner/ResumingStreamingResultScannerTest.java

```diff
@@ -261,4 +306,72 @@ private void doErrorsDoNotResume(Status status) throws IOException {
     verifyNoMoreInteractions(mockScannerPostResume, mockScannerFactory);
     scanner.close();
   }
+
+  @Test
+  /**
+   * Test successfully retrying while receiving a mixture of timeouts and retryable exceptions
+   */
+  public void testTimeoutsMixedWithErrors() throws IOException {
+    Row row1 = buildRow("row1");
+    Row row2 = buildRow("row2");
+    Row row3 = buildRow("row3");
+    Row row4 = buildRow("row4");
+
+    IOException retryableException =
+        new IOExceptionWithStatus("Test", new StatusRuntimeException(Status.UNAVAILABLE));
+    IOException timeoutException = new ScanTimeoutException("Test");
+
+    ReadRowsRequest.Builder originalRequest = readRowsRequest.toBuilder();
+
+   when(mockScannerFactory.createScanner(eq(originalRequest.build())))
+        .thenReturn(mockScanner);
+    ResultScanner<Row> afterError1Mock = mock(ResultScanner.class);
+    ResultScanner<Row> afterTimeoutMock = mock(ResultScanner.class);
+    ResultScanner<Row> afterError2Mock = mock(ResultScanner.class);
+
+    ReadRowsRequest.Builder expectedResumeRequest = originalRequest.build().toBuilder();
+    expectedResumeRequest.getRowRangeBuilder()
+        .setStartKey(ResumingStreamingResultScanner.nextRowKey(ByteString.copyFromUtf8("row1")));
```

Comment:
Can you please add a variable for ResumingStreamingResultScanner.nextRowKey(ByteString.copyFromUtf8("row1")) to break up the long command?